checkov/sast/checks/checks_parser.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+import logging
+from typing import Dict, Any, List
+
+from checkov.common.graph.checks_infra.enums import Operators
+from checkov.sast.consts import COMPARISON_VALUES, COMPARISON_VALUE_TO_SYMBOL, SemgrepAttribute, \
+    CHECKOV_SEVERITY_TO_SEMGREP_SEVERITY, PATTERN_OPERATOR_TO_SEMGREP_ATTR, VARIABLE_OPERATOR_TO_SEMGREP_ATTR, \
+    FILTER_OPERATOR_TO_SEMGREP_ATTR, BqlConditionType
+from checkov.common.util.type_forcers import force_list
+
+
+class SastCheckParser:
+    def parse_raw_check_to_semgrep(self, raw_check: Dict[str, Dict[str, Any]]) -> Dict[str, Any]:
+        semgrep_rule: Dict[str, Any] = {}
+        if not self.raw_check_is_valid(raw_check):
+            logging.error(f'cant parse the following policy: {raw_check}')
+        try:
+            semgrep_rule = self.parse_rule_metadata(raw_check, semgrep_rule)
+            semgrep_rule.update(self.parse_definition(raw_check['definition']))
+        except Exception as e:
```

Comment:
what's the point of except here if you just raise it back?
Also, I'm not sure how it works in new python versions (I mainly used 2.7 unfortunately in the past), but I think there is a different behavior between letting the exception be raised or catching it and using `raise e`, as it overrides the stacktrace. Instead you can just use `raise` (without the `e`) and it makes sure to not create a new object/stacktrace